cluster/node-pools/worker-karpenter/provisioners.yaml

```diff
@@ -0,0 +1,121 @@
+{{ if eq .Cluster.ConfigItems.karpenter_pools_enabled "true"}}
+---
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: "{{.NodePool.Name}}"
+spec:
+  {{- if index .NodePool.ConfigItems "scaling_priority"}}
+  weight: {{.NodePool.ConfigItems.scaling_priority}}
+  {{- end}}
+  consolidation:
+    {{- if index .NodePool.ConfigItems "consolidation" }}
+    enabled: {{ .NodePool.ConfigItems.consolidation }}
+    {{- else }}
+    enabled: true
+    {{- end }}
+  requirements:
+    - key: "node.kubernetes.io/instance-type"
+      operator: In
+      values:
+{{- range $type := .NodePool.InstanceTypes }}
+      - "{{ $type }}"
+{{- end }}
+    - key: karpenter.sh/capacity-type
+      operator: In
+      values:
+        - "spot"
+        - "on-demand"
+    - key: "kubernetes.io/arch"
+      operator: In
+      values: ["arm64", "amd64"]
+    - key: "topology.kubernetes.io/zone"
+      operator: In
+      values:
+{{- $azs := .Values.availability_zones }}
+{{- if index .NodePool.ConfigItems "availability_zones"}}
+        {{    $azs = split .NodePool.ConfigItems.availability_zones "," }}
+{{- end}}
+{{- range $az :=  $azs }}
+      - "{{ $az }}"
+{{- end }}
+{{- $taints := .NodePool.Taints}}
+{{- if $taints }}
+  taints:
+  {{- range $taints }}
+    {{- $taint := . }}
+      - key: {{$taint.Key}}
+        effect: {{$taint.Effect}}
+        {{- if $taint.Value }}
+        value: {{$taint.Value }}
+        {{- end }}
+  {{- end}}
+{{- end}}
+
+  startupTaints:
+  - key: zalando.org/node-not-ready
+    effect: NoSchedule
+
+  labels:
+    # these labels are normally set by kubelet on start-up, but because
+    # karpenter already creates the node object ahead of the instance start, we
+    # need to set them here as well.
+    lifecycle-status: ready
+    node.kubernetes.io/node-pool: "{{ .NodePool.Name }}"
+    node.kubernetes.io/role: worker
+    node.kubernetes.io/profile: "{{ .NodePool.Profile }}"
+    cluster-lifecycle-controller.zalan.do/replacement-strategy: none
+{{- if index .NodePool.ConfigItems "labels"}}
+  {{- range split .NodePool.ConfigItems.labels ","}}
+    {{- $label := split . "="}}
+    {{index $label 0}}: {{index $label 1}}
+  {{- end}}
+{{- end}}
+
+  # ttlSecondsAfterEmpty: 30
+  providerRef:
+    name: {{ .NodePool.Name }}
+---
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: {{ .NodePool.Name }}
+spec:
+  amiFamily: Custom
+  amiSelector:
+    aws-ids: "{{ .Cluster.ConfigItems.kuberuntu_image_v1_22_amd64  }},{{ .Cluster.ConfigItems.kuberuntu_image_v1_22_arm64 }}"
+  metadataOptions:
+    httpTokens: optional
+  subnetSelector:
+    Name: "dmz-eu-central-1*" # TODO: use a better tag + it has to have auto assign public IP enabled
```

Comment:
I assume fixing the public IP settings (either on subnet or in karpenter) is sort of important for merging this?

Reply:
yes it is, for teapot cluster I switched it on manually after consulting with linus. for wider rollout we need them to enable it or we do it in karpenter and update the version here

Reply:
Ok, it doesn't have to be a blocker for merging this IMO, it just means we can only realistically test in dev clusters until it's resolved. But it's fine to fix it later in a follow up.

Reply:
this was resolved for dev clusters
https://github.bus.zalan.do/Base-Infrastructure/aws-network-infrastructure/pull/448/files